aten/src/ATen/xpu/XPUEvent.h

```diff
@@ -0,0 +1,145 @@
+#pragma once
+#include <ATen/xpu/XPUContext.h>
+#include <c10/util/Optional.h>
+
+namespace at::xpu {
+
+/*
+ * XPUEvent are movable not copyable wrappers around SYCL event. XPUEvent are
+ * constructed lazily when first recorded. It has a device, and this device is
+ * acquired from the first recording stream. Later streams that record the event
+ * must match the same device.
+ *
+ * Currently, XPUEvent does NOT support to export an inter-process event from
+ * another process via inter-process comunication(IPC). So it means that
+ * inter-process communication for event handles between different processes is
+ * not available. This could impact some applications that rely on cross-process
+ * synchronization and communication.
+ */
+struct TORCH_XPU_API XPUEvent {
+  // Constructors
+  XPUEvent() noexcept = default;
+  XPUEvent(bool enable_timing) noexcept : enable_timing_{enable_timing} {}
+
+  ~XPUEvent() {}
+
+  XPUEvent(const XPUEvent&) = delete;
+  XPUEvent& operator=(const XPUEvent&) = delete;
+
+  XPUEvent(XPUEvent&& other) noexcept {
+    moveHelper(std::move(other));
+  }
+
+  XPUEvent& operator=(XPUEvent&& other) noexcept {
+    if (this != &other) {
+      moveHelper(std::move(other));
+    }
+    return *this;
+  }
+
+  operator sycl::event&() const {
+    return event();
+  }
+
+  optional<at::Device> device() const {
+    if (isCreated()) {
+      return at::Device(at::kXPU, device_index_);
+    } else {
+      return {};
+    }
+  }
+
+  bool isCreated() const {
+    return (event_.get() != nullptr);
```

Comment:
Nit (replies on https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool )
```suggestion
    return event_;
```

Reply:
This will raise this error: 
```CPP
error: cannot convert ‘const std::unique_ptr<sycl::_V1::event>’ to ‘bool’ in return
```
std::unique_ptr is not implicitly convertible to bool. It is contextually convertible to bool (due to its [explicit conversion operator](http://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool)), which is why you can use it in an if statement, or put a ! in front of it, but you cannot pass it as an argument to a function which is expecting a bool.